pkg/kubelet/eviction/helpers_test.go

```diff
@@ -1319,11 +1322,25 @@ func TestMakeSignalObservations(t *testing.T) {
 	if !found {
 		t.Error("Expected available memory observation")
 	}
-	if expectedBytes := int64(nodeAvailableBytes); memQuantity.available.Value() != expectedBytes {
-		t.Errorf("Expected %v, actual: %v", expectedBytes, memQuantity.available.Value())
-	}
-	if expectedBytes := int64(nodeWorkingSetBytes + nodeAvailableBytes); memQuantity.capacity.Value() != expectedBytes {
-		t.Errorf("Expected %v, actual: %v", expectedBytes, memQuantity.capacity.Value())
+	if runtime.GOOS == "windows" {
+		// on Windows evictionapi.SignalMemoryAvailable is constructed from global memory stats fetched from system perf counters
+		availablePhysMem, totalPhysMem, err := winstats.GetAvailableAndTotalPhysicalMemory()
+		if err != nil {
+			t.Errorf("Error fetching available and total physical memory: %v", err)
+		}
+		if expectedBytes := int64(totalPhysMem); memQuantity.capacity.Value() != expectedBytes {
+			t.Errorf("Expected %v, actual: %v", expectedBytes, memQuantity.capacity.Value())
+		}
+		if math.Abs(float64(int64(availablePhysMem)-memQuantity.available.Value())) > (10 * 1024 * 1024) {
+			t.Errorf("Reported available memory is not within 10Mb of expected value. Expected: %v, actual: %v", availablePhysMem, memQuantity.available.Value())
+		}
```

Comment:
This is to handle fluctuation in the memory since it is a queried value?  Do we need to take into account the other direction (where memory might have been freed)?

Reply:
This checks the absolute value of the delta so i think that should cover if more memory has been freed too?

Reply:
thanks I missed that!